api-put-object-progress.go

```diff
@@ -121,3 +121,51 @@ func (c Client) PutObjectWithMetadata(bucketName, objectName string, reader io.R
 	}
 	return n, nil
 }
+
+// PutObjectStreaming using AWS streaming signature V4
+func (c Client) PutObjectStreaming(bucketName, objectName string, reader io.Reader, size int64) (n int64, err error) {
```

Comment:
I have added an example under `examples/s3/putobject-streaming.go` and updated `docs/API.md`